qa/tasks/scripts.py

```diff
@@ -0,0 +1,394 @@
+_initialize_minion_configs_array $DESTDIR
+for DESTFILE in "${MINION_CONFIGS_ARRAY[@]}" ; do
+    cp $SOURCEFILE $DESTDIR/$DESTFILE
+done
+echo "Your custom storage profile $SOURCEFILE has the following contents:"
+cat $DESTDIR/$DESTFILE
+ls -lR $PROPOSALSDIR/profile-custom
+""",
+        "mgr_dashboard_module_smoke": """# smoke test the MGR dashbaord module
+set -ex
+URL=$(ceph mgr services 2>/dev/null | jq .dashboard | sed -e 's/"//g')
+curl --insecure --silent $URL 2>&1 > dashboard.html
+test -s dashboard.html
+file dashboard.html | grep "HTML document"
+echo "OK" >/dev/null
+""",
+        "ceph_version_sanity": """# Ceph version sanity test
+# test that ceph RPM version matches "ceph --version"
+# for a loose definition of "matches"
+set -ex
+rpm -q ceph
+RPM_NAME=$(rpm -q ceph)
+RPM_CEPH_VERSION=$(perl -e '"'"$RPM_NAME"'" =~ m/ceph-(\d+\.\d+\.\d+)/; print "$1\n";')
+echo "According to RPM, the ceph upstream version is ->$RPM_CEPH_VERSION<-" >/dev/null
+test -n "$RPM_CEPH_VERSION"
+ceph --version
+BUFFER=$(ceph --version)
+CEPH_CEPH_VERSION=$(perl -e '"'"$BUFFER"'" =~ m/ceph version (\d+\.\d+\.\d+)/; print "$1\n";')
+echo "According to \"ceph --version\", the ceph upstream version is ->$CEPH_CEPH_VERSION<-" \
+    >/dev/null
+test -n "$RPM_CEPH_VERSION"
+test "$RPM_CEPH_VERSION" = "$CEPH_CEPH_VERSION"
+""",
+        "rados_write_test": """Write a RADOS object and read it back
+#
+# NOTE: function assumes the pool "write_test" already exists. Pool can be
+# created by calling e.g. "create_all_pools_at_once write_test" immediately
+# before calling this function.
+#
+set -ex
+ceph osd pool application enable write_test deepsea_qa
+echo "dummy_content" > verify.txt
+rados -p write_test put test_object verify.txt
+rados -p write_test get test_object verify_returned.txt
+test "x$(cat verify.txt)" = "x$(cat verify_returned.txt)"
+""",
+        "mgr_plugin_influx": """# Influx MGR plugin smoke test
+set -ex
+influx -version
+systemctl start influxdb.service
+sleep 5
+systemctl status --full --lines=0 influxdb.service
+influx -execute 'create database ceph'
+influx -database ceph \
+    -execute "create user admin with password 'badpassword' with all privileges"
+ceph mgr module enable influx
+ceph influx config-set hostname $(hostname)
+ceph influx config-set port 8086
+ceph influx config-set username admin
+ceph influx config-set password badpassword
+ceph influx config-set database ceph
+ceph influx config-set ssl false
+ceph influx config-set verify_ssl false
+ceph influx send
+sleep 5
+ceph -s
+sleep 5
+influx -database ceph -execute 'show series' | head -n1 | grep key
+influx -database ceph -execute 'select * from ceph_daemon_stats limit 1' | \
+    head -n1 | grep ceph_daemon_stats
+echo "OK" >/dev/null
+""",
+        }
+
+    def __init__(self, master_remote, logger):
+        self.log = logger
+        self.master_remote = master_remote
+
+    def ceph_cluster_status(self, *args, **kwargs):
```

Comment:
All those functions is an overkill to have. Why you need proxy method for all of them and just don't use remote_run_script_as_root directly?

Reply:
I'm going to get rid of these functions soon.